### PR TITLE
OAuth implementation

### LINKED ISSUE #225: OAuth
Password authentication is being removed in march of 2017.

### PR DESCRIPTION
**Pull request by Innectic**
Fixes #225

### CODE REVIEW

File: cactusbot/cactus.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def __init__(self, service):
 
         self.service = service
 
-    async def run(self, api, *auth):
```
Comment (2Cubed) [inline cactusbot/cactus.py]: This should stay as `*auth` - remember, service agnostic. (Discord needs two parameters, for example.)

File: cactusbot/cactus.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def __init__(self, service):
...
+    async def run(self, api, token):
```
Comment (2Cubed) [inline cactusbot/cactus.py]: Where is `token` actually going? Doesn't look like it's being used... o.O

File: cactusbot/handler.py [hunk 3]
```diff
@@ -13,6 +13,10 @@ def __init__(self, *handlers):
 
         self.handlers = handlers
 
+    async def register_handler(self, handler):
```
Comment (2Cubed) [inline cactusbot/handler.py]: This won't work; `handlers` is a `tuple`, and you're trying to `+=` something to it. (Test your code! >.>)

What might be better is to just make `self.handlers = list(handlers)` in `__init__`, and then just use `handlers.handlers.append(new_handler_thing)`.

File: caches/followers.json [hunk 4]
```diff
@@ -1 +1,3 @@
-{}
+{
+  "xIGBClutchIx": "2017-03-03T18:02:31.828988"
+}
```
Comment (2Cubed) [inline caches/followers.json]: \>.>

File: cactusbot/handlers/respond.py [hunk 5]
```diff
@@ -6,10 +6,15 @@
...
-    def __init__(self, username):
+    def __init__(self):
         super().__init__()
 
-        self.username = username
+        self.username = ""
+
+    async def on_username_update(self, packet):
+        """Set the username of the bot."""
+        print(packet.json)
```
Comment (2Cubed) [inline cactusbot/handlers/respond.py]: <.<

**Review by 2Cubed — CHANGES_REQUESTED** [review]

File: cactusbot/handler.py (hunk 3 above)
Comment (Innectic) [inline cactusbot/handler.py]: I forgot to remove that, it's not actually being used. I thought of that while I was writing that, and changed it to an event.

File: caches/followers.json (hunk 4 above)
Comment (Innectic) [inline caches/followers.json]: Added this to the gitignore ;P

**Review by 2Cubed — APPROVED** [review]

**Review by ProbablePrime — COMMENTED** [review]
ya thatll do for now

**Comment by 2Cubed** [comment]
@ProbablePrime Awesome. Will this be changing, though?

**Comment by ProbablePrime** [comment]
No changes will affect this, implicit oauth here to stay

**Pull request merged**